test/net_http_test.rb

```diff
@@ -0,0 +1,389 @@
+require 'test_helper'
+require 'semian/net_http'
+require 'thin'
+
+class TestNetHTTP < MiniTest::Unit::TestCase
+  class RackServer
+    def self.call(env)
+      response_code = env['REQUEST_URI'].delete("/")
+      response_code = '200' if response_code == ""
+      [response_code, {'Content-Type' => 'text/html'}, ['Success']]
+    end
+  end
+
+  PORT = 31_050
+  TOXIC_PORT = PORT + 1
+
+  DEFAULT_SEMIAN_OPTIONS = {
+    tickets: 3,
+    success_threshold: 1,
+    error_threshold: 3,
+    error_timeout: 10,
+  }
+
+  def test_semian_identifier
+    with_server do
```

Comment:
Pass the `port` of the server as an argument to the block. This will easily allow you to create a new test where multiple servers are alive and you query them. Or even better, pass the entire hostname + port it is set up for (see my comment on `with_server` to make it reject other connections):

```ruby
with_server do |hostname|
  open_circuit!(hostname)
  Net::HTTP.get(URI(hostname))
end
```

And so on :-)